src/components/d3/CirclesGroup.js

```diff
@@ -0,0 +1,66 @@
+const React = require('react');
+
+const StyleConstants = require('../../constants/Style');
+
+const CirclesGroup = React.createClass({
+  propTypes: {
+    adjustedHeight: React.PropTypes.number.isRequired,
+    circleRadius: React.PropTypes.number,
+    data: React.PropTypes.array.isRequired,
+    shouldAnimate: React.PropTypes.bool,
+    translation: React.PropTypes.string,
+    xScaleValueFunction: React.PropTypes.func.isRequired,
+    yScaleValueFunction: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      circleRadius: 3,
+      shouldAnimate: true,
+      translation: 'translate(0,0)'
+    };
+  },
+
+  componentDidMount () {
+    this._animateCircles();
+  },
+
+  componentDidUpdate () {
+    this._animateCircles();
+  },
+
+  _animateCircles () {
+    if (this.props.shouldAnimate) {
+      d3.select(this.refs.circleGroup).selectAll('.circle').data(this.props.data).transition().attr('cy', d => {
+        return this.props.yScaleValueFunction(d.y);
+      });
+    }
+  },
+
+  render () {
+    return (
+      <g className='circle-group' ref='circleGroup' transform={this.props.translation}>
+        {this.props.data.length <= 45 ? (
```

Comment:
So you cannot pass an array that has more than 45 elements? What is the limitation here?